packages/fx-core/src/common/featureFlags.ts

```diff
@@ -81,3 +79,49 @@ export function isNewProjectTypeEnabled(): boolean {
   return isFeatureFlagEnabled(FeatureFlagName.OfficeAddin, false);
 }
+
+///////////////////////////////////////////////////////////////////////////////
+// Notes for Office Addin Feature flags:
+// Case 1: TEAMSFX_OFFICE_ADDIN = false, TEAMSFX_OFFICE_XML_ADDIN = false
+//   1.1 project-type option: `outlook-addin-type`
+//   1.2 addin-host: not show but use `outlook` internally
+//   1.3 capabilities options: [`jason-taskpane`, `outlook-addin-import`]
+//   1.4 programming-language options: [`typescript`] (skip in UI)
+//   1.5 office-addin-framework-type: not show question but use `default_old` internally
+//   1.6 generator class: OfficeAddinGenerator
+//   1.7 template link: config.json.json-taskpane.default_old.typescript
+// Case 2: TEAMSFX_OFFICE_ADDIN = false AND TEAMSFX_OFFICE_XML_ADDIN = true
+//   2.1 project-type option: `office-xml-addin-type`
+//   2.2 addin-host options: [`outlook`, `word`, `excel`, `powerpoint`]
+//   2.3 capabilities options:
+//     if (addin-host == `outlook`) then [`json-taskpane`, `outlook-addin-import`]
+//     else if (addin-host == `word`) then [`word-taskpane`, `word-xxx`, ...]
+//     else if (addin-host == `excel`) then [`excel-taskpane`, `excel-xxx`, ...]
+//     else if (addin-host === `powerpoint`) then [`powerpoint-taskpane`, `powerpoint-xxx`, ...]
+//   2.4 programming-language options:
+//     if (addin-host == `outlook`) then [`typescript`] (skip in UI)
+//     else two options: [`typescript`, `javascript`]
+//   2.5 office-addin-framework-type options: not show but use `default` internally
+//   2.6 generator class:
+//     if (addin-host == `outlook`) then OfficeAddinGenerator
+//     else OfficeXMLAddinGenerator
+//   2.7 template link:
+//     if (addin-host == `outlook`) config.json.json-taskpane.default.[programming-language]
+//     else config[addin-host].[capabilities].default.[programming-language]
+// Case 3: TEAMSFX_OFFICE_ADDIN = true AND TEAMSFX_OFFICE_XML_ADDIN = true
+//     3.1 project-type option: `office-addin-type`
+//   3.2 addin-host: not show but will use `wxpo` internally
+//   3.3 capabilities options: [`json-taskpane`, `office-addin-import`]
+//   3.4 programming-language options: [`typescript`, `javascript`]
+//   3.5 office-addin-framework-type options: [`default`, `react`]
+//   3.6 generator class: OfficeAddinGenerator
+//   3.7 template link: config.json.json-taskpane.[office-addin-framework-type].[programming-language]
+// case 4: TEAMSFX_OFFICE_ADDIN = true AND TEAMSFX_OFFICE_XML_ADDIN = fasle
+//   4.1 project-type option: `office-addin-type`
+//   4.2 addin-host: not show but will use `wxpo` internally
+//   4.3 capabilities options: [`jason-taskpane`, `office-addin-import`]
+//   4.4 programming-language options: [`typescript`, `javascript`]
+//   4.5 office-addin-framework-type options: [`default`, `react`]
+//   4.6 generator class: OfficeAddinGenerator
+//   4.7 template link: config.json.json-taskpane.[office-addin-framework-type].[programming-language]
```

Comment:
Thanks for the greate summary here.

Since this two case have the same sub items,\
maybe we can combine them into `TEAMSFX_OFFICE_ADDIN = true` && no matter what `TEAMSFX_OFFICE_XML_ADDIN`'s value\
to shorter the comments here?